### DISCUSSION TITLE
Adding ROS2 mesage parsing capabilities to the rust mcap crate

### DISCUSSION THREAD

**Discussion by A-K-O-R-A** [foxglove/mcap#1193] [General]
The python package [[1]](https://mcap.dev/docs/python/ros2_example) and go cli have the capabilities of decoding the ROS messages contained in the parsed MCAP files. As of now the rust crate does not contain a functionality to do so or links another crate that allows developers to easily implement this. Overall this makes it hard to create tools with mcap files in rust. My question is if ros2 message decoding is outside of the scope of the crate or should be included into the crate or at least mentioned in examples in combination with another crate.

**Comment by jtbandes** [comment]
I'm not sure if there is any prior art for decoding ros2/cdr messages dynamically in Rust. A quick google search turns up your project as the top result 😄

**Reply by A-K-O-R-A** [reply]
Yes, I recently needed to use mcap files in rust and implemented it myself. I totally understand if adding it as dependency is not an option for security reasons, everything is MIT though so maintaining this as a foxglove owned package would not be a problem besides the maintenance effort. I was just wondering if integrating this functionality into the mcap package is considered as an option.

The problem with my package is that as of now I do not have enough test data to test out the different types of CDR which makes it hard to asses the robustness of my implementation and for my own use it doesn't really matter. If you do consider adding it then I'd be happy to spend more time on polishing the crate though :)

**Reply by defunctzombie** [reply]
Traditionally my main feedback about adding specific serialization support to any of our "mcap" packages has been dependency bloat for users that don't need these. However, from what I know about Rust crates - there's a nice "feature" system that lets folks gracefully opt-into having or not having specific features and dependencies. If there's a way to leverage that I think there's room to entertain a proposal here.

cc @james-rms @eloff

**Reply by eloff** [reply]
One could definitely put it behind a feature flag
